packages/react-components/react-carousel-preview/src/components/CarouselButton/useCarouselButton.tsx

```diff
@@ -16,26 +19,27 @@ import { useCarouselValues_unstable } from '../useCarouselValues';
  */
 export const useCarouselButton_unstable = (
   props: CarouselButtonProps,
-  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+  ref: React.Ref<ARIAButtonElement>,
 ): CarouselButtonState => {
   const { navType } = props;
 
   const selectPageByDirection = useCarouselContext_unstable(c => c.selectPageByDirection);
   const values = useCarouselValues_unstable(snapshot => snapshot);
   const activeValue = useCarouselContext_unstable(c => c.value);
+  const loop = useCarouselContext_unstable(c => c.loop);
 
   const handleClick = (event: React.MouseEvent<HTMLButtonElement & HTMLAnchorElement>) => {
     if (event.isDefaultPrevented()) {
       return;
     }
 
-    selectPageByDirection(event, navType);
+    selectPageByDirection(event, navType, loop);
```

Comment:
```suggestion
    selectPageByDirection(event, navType);
```

As `useCarousel()` knows about `loop` value.